### ISSUE TITLE
Init Container support

### ISSUE THREAD

**Issue by ztittle** [dotnet/tye#637]
### What should we add or change to make your life better?

Support the [Init Container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) pattern to allow running containerized setup scripts before a given microservice container starts.

### Why is this important to you?

This is useful for us since we have an init container running in production that dynamically sets up a configuration file that is needed by the microservice before the pod starts. Without init container support, we would need to add a wrapper script around tye in order for it to startup correctly for a local environment.

**Comment by glennc** [comment]
Do you also run these scripts as part of the dev flow? i.e should this be part of this proposal: https://github.com/dotnet/tye/issues/627

**Comment by ztittle** [comment]
Yes. After looking at the proposal, I think it would satisfy the need we have.
